source/logrepl/combined.go

```diff
@@ -134,6 +135,77 @@ func (c *CombinedIterator) Next(ctx context.Context) (opencdc.Record, error) {
 	return r, nil
 }
 
+// NextN retrieves up to n records from the active iterator.
+// If the end of the snapshot is reached during this call, it will switch to the CDC iterator
+// and continue retrieving records from there.
+func (c *CombinedIterator) NextN(ctx context.Context, n int) ([]opencdc.Record, error) {
+	if n <= 0 {
+		return nil, fmt.Errorf("n must be greater than 0, got %d", n)
+	}
+
+	// Check if the active iterator already implements NextN (like the CDCIterator)
+	if nextN, ok := c.activeIterator.(interface {
+		NextN(context.Context, int) ([]opencdc.Record, error)
+	}); ok {
+		records, err := nextN.NextN(ctx, n)
+		if err != nil {
+			// Snapshot iterator is done, handover to CDC iterator
+			if errors.Is(err, snapshot.ErrIteratorDone) {
+				if err := c.useCDCIterator(ctx); err != nil {
+					return nil, err
+				}
+				sdk.Logger(ctx).Debug().Msg("Snapshot completed, switching to CDC mode")
+
+				// Retry with new iterator
+				return c.NextN(ctx, n)
+			}
+			return nil, fmt.Errorf("failed to fetch records in batch: %w", err)
+		}
+		return records, nil
+	}
+
+	// Fall back to using Next() if NextN is not implemented
```

Comment:
I'd simply remove `Next` and not make our lives harder than they are. This code is internal to the connector, so we're not breaking anything.

Also, since `CombinedIterator` just forwards the call to either the `CDCIterator` or the `SnapshotIterator`, I don't think we need the code to fetch the remaining (n-1) records, the actual iterator will already do that for us.